tasks.c

```diff
@@ -7653,3 +7653,52 @@ static void prvAddCurrentTaskToDelayedList( TickType_t xTicksToWait,
     #endif
 
 #endif /* if ( configINCLUDE_FREERTOS_TASK_C_ADDITIONS_H == 1 ) */
+/*-----------------------------------------------------------*/
+
+#if ( ( configSUPPORT_STATIC_ALLOCATION == 1 ) && ( configKERNEL_PROVIDED_STATIC_MEMORY == 1 ) )
+
+/*
+ * This is the kernel provided implementation of vApplicationGetIdleTaskMemory()
+ * to provide the memory that is used by the Idle task. It is used when
+ * configKERNEL_PROVIDED_STATIC_MEMORY is set to 1. The application can provide
+ * it's own implementation of vApplicationGetIdleTaskMemory by setting
+ * configKERNEL_PROVIDED_STATIC_MEMORY to 0 or leaving it undefined.
+ */
+    void vApplicationGetIdleTaskMemory( StaticTask_t ** ppxIdleTaskTCBBuffer,
+                                        StackType_t ** ppxIdleTaskStackBuffer,
+                                        uint32_t * pulIdleTaskStackSize )
+    {
+        static StaticTask_t xIdleTaskTCB;
+        static StackType_t uxIdleTaskStack[ configMINIMAL_STACK_SIZE ];
```

Comment:
Would it make sense to align these to the region size for MPU ports?
It becomes wasted memory on the ArmV8-M MPU ports, as they only require [32 byte alignment](https://developer.arm.com/documentation/107565/0100/Introduction/Differences-between-Armv7-M-and-Armv8-M-MPU), but it would make this usable for the ArmV7-M MPU ports, as they require MPU region alignment to the size of the region. 
I think doing this would allow these functions to be used by all of our ports

Reply:
These have been made part of privileged data and that section is aligned using linker scripts. There is no need to any explicit alignment.

Reply:
Well the task stack will be turned into an MPU region, so if that's not aligned correctly it can lead to issues depending on the architecture.
I know on the ArmV7-R that having an unaligned MPU region can lead to "[Unpredictable Behaviour](https://developer.arm.com/documentation/ddi0363/g/Memory-Protection-Unit/About-the-MPU/Memory-regions?lang=en)"
And it looks like on the ArmV7-M that  "[Any unaligned access to Device or Strongly Ordered memory generates an alignment UsageFault](https://developer.arm.com/documentation/ddi0489/f/memory-system/tcm-interfaces/tcm-attributes-and-permissions)". 

Where right now the `portPRIVILEGED_RAM_REGION` [MPU region over-writes](https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/V10.6.0/portable/GCC/ARM_CM3_MPU/portmacro.h#L88) the `portSTACK_REGION`, where I don't think this is an issue, as you stated.

*However*, if we ever set `portSTACK_REGION` to have a higher priority than `portPRIVILEGED_RAM_REGION`, which would allow dynamic creation of unprivileged tasks, this could potentially be a pretty tough issue to debug.

All that said, I don't think this is something worth holding up this PR over.

Reply:
Yes, you are right. Stack memory needs to satisfy alignment requirements as we use an MPU region to guard that. I think it is best to supply these default implementations for non-MPU ports as alignment requirements very with the architecture.